sys/net/sock/sock_util.c

```diff
@@ -188,6 +188,17 @@ int _parse_netif(sock_udp_ep_t *ep_out, char *netifstart)
     return (netifend - netifstart);
 }
 
+int _get_default_netif(sock_udp_ep_t *ep_out)
+{
+    netif_t *netif = netif_iter(NULL);
+    if (netif == NULL || netif_iter(netif) != NULL) {
```

Comment:
Why checking again for `netif_iter(netif) != NULL` ? According to doc `netif_iter` returns `NULL, if there is no interface after @p last`, which would be already true for the first call of `netif_iter`.

Reply:
- `netif_iter(NULL)` gives the first interface
 - `netif_iter(netif_iter(NULL))` gives the second interface
 - and so on

This should only set the interface if there is a first, but not a second interface.